balloon/Cargo.toml

```diff
@@ -0,0 +1,35 @@
+[package]
+name = "balloon-hash"
+version = "0.1.0" # Also update html_root_url in lib.rs when bumping this
+description = "Pure Rust implementation of the Balloon password hashing function"
+authors = ["RustCrypto Developers"]
+license = "MIT OR Apache-2.0"
+documentation = "https://docs.rs/balloon-hash"
+repository = "https://github.com/RustCrypto/password-hashes/tree/master/balloon"
+keywords = ["crypto", "password", "hashing"]
+categories = ["cryptography", "no-std"]
+edition = "2018"
+readme = "README.md"
+
+[dependencies]
+digest = { version = "0.9", default-features = false }
+crypto-bigint = { version = "0.2.10", default-features = false, features = ["generic-array"] }
+
+# optional dependencies
+password-hash = { version = "0.3", default-features = false, optional = true }
+rayon = { version = "1", optional = true }
+
+[dev-dependencies]
+nachonavarro-balloon = { git = "https://github.com/khonsulabs/nachonavarro-balloon" }
```

Comment:
Is having a git `dev-dependency` acceptable? Alternatives:
- Pin to a specific commit
- Release it on crates.io
- Move it's implementation inside of this crate

Ofc we are also willing to donate the crate to RustCrypto if it's preferred.

Reply:
We can publish a crate with a `git` dev dependency, but the real downsides are it's outside our org and can break the build if it's in a bad state

Reply:
As I said, we are willing to move the crate to the RustCrypto org, feel free to hit me up anytime for it.

Reply:
I just added it right into the testing facilities, is that alright? It's easy to make it a separate crate, just say the word.

Also this adds a git submodule, I hope that's alright? If not, any advice on what to do instead would be appreciated. Just copying the file seemed problematic to me because of attribution and what not.

Reply:
I also added automatic updates to the gitsubmodule via dependabot.

Reply:
I agree, I'm unsure how to proceed though. What would you recommend?

> If not, any advice on what to do instead would be appreciated. Just copying the file seemed problematic to me because of attribution and what not.

Reply:
If it's OSS, and only for testing, I don't see an issue.

Vendor it into the repo and if need be I can work on the proper boilerplate for attribution.

Reply:
I think it should be enough to simply generate a bunch of test vectors using `nachonavarro-balloon` and test against them instead of depending on it. We can simply leave a comment stating that the test vectors were generated using this library.

Reply:
Done. I hope the test vectors already generated are enough, if not let me know!